src/rastervision/core/ml_backend.py

```diff
@@ -9,9 +9,21 @@ class MLBackend(ABC):
     """
 
     @abstractmethod
-    def convert_training_data(self, training_data, validation_data, class_map,
-                              options):
-        """Convert training data to backend-specific format and save it.
+    def per_project_data_processor(self, project, data, class_map, options):
+        """Process each project's training data
+
+        Args:
+            project: Project
+            data: TrainingData
+            class_map: ClassMap
+            options: ProcessTrainingDataConfig.Options
+        """
+        pass
+
+    @abstractmethod
+    def all_projects_dataset_processor(self, training_data, validation_data,
+                                       class_map, options):
+        """After all projects have been processed, process the resultset
```

Comment:
The name `training_data` is confusing since that name is used in other places to refer to objects of type `TrainingData`. In this case, it's used to refer to some backend-specific way of representing paths to project datasets. I can see that there's no good concise way to encapsulate this in a variable name, but I think it would be helpful to explain the contents of this variable in the Pydocs, both for the abstract methods and the implementations in the backend classes.